crates/red_knot_python_semantic/src/semantic_index/re_exports.rs

```diff
@@ -0,0 +1,299 @@
+//! A visitor and query to find all global-scope symbols that are exported from a module
+//! when a wildcard import is used.
+//!
+//! For example, if a module `foo` contains `from bar import *`, which symbols from the global
+//! scope of `bar` are imported into the global namespace of `foo`?
+
+use ruff_db::{files::File, parsed::parsed_module};
+use ruff_python_ast::{
+    self as ast,
+    name::Name,
+    visitor::{walk_expr, walk_pattern, walk_stmt, Visitor},
+};
+use rustc_hash::FxHashSet;
+
+use crate::{module_name::module_name_from_import_statement, resolve_module, Db};
+
+#[salsa::tracked(return_ref)]
+pub(super) fn find_exports(db: &dyn Db, file: File) -> FxHashSet<Name> {
```

Comment:
Nit: `find_exports` is a somewhat unusual name for a salsa query. We normally name these after what they return rather than using a verb. Maybe just `exports` or `exported_names`?